futures-util/src/future/map.rs

```diff
@@ -9,40 +8,37 @@ use futures_core::task;
 /// This is created by the `Future::map` method.
 #[derive(Debug)]
 #[must_use = "futures do nothing unless polled"]
-pub struct Map<A, F> where A: Future {
-    future: A,
-    f: Option<F>,
+pub struct Map<Fut, F> {
+    future: Fut,
+    op: Option<F>,
 }
 
-pub fn new<A, F>(future: A, f: F) -> Map<A, F>
-    where A: Future,
-{
-    Map {
-        future,
-        f: Some(f),
-    }
-}
+impl<Fut, F> Map<Fut, F> {
+    unsafe_pinned!(future -> Fut);
+    unsafe_unpinned!(op -> Option<F>);
 
-impl<A: Future, F> Map<A, F> {
-    unsafe_pinned!(future -> A);
-    unsafe_unpinned!(f -> Option<F>);
+    /// Creates a new Map.
+    pub fn new(future: Fut, op: F) -> Map<Fut, F> {
```

Comment:
The previous `new` functions where actually not exported publicly. This change makes the constructors public and thus exposes what types are used internally. If moved from module functions to inherent functions, they should be `pub(super)`.